packages/node/src/login/oidc/refresh/TokenRefresher.spec.ts

```diff
@@ -24,44 +24,279 @@ import {
+  }
+  const mockedIssuer = {
+    metadata: mockDefaultIssuerConfig(),
+    Client: clientConstructor,
+  };
+  Issuer.mockReturnValueOnce(mockedIssuer);
+};
+
+const setupDefaultOidcClientMock = () => {
+  setupOidcClientMock(mockDpopTokens());
+};
+
+const mockRefresherDefaultStorageUtility = () =>
+  mockStorageUtility({
+    "solidClientAuthenticationUser:mySession": {
+      issuer: "https://my.idp",
+      codeVerifier: "some code verifier",
+      redirectUri: "https://my.app/redirect",
+      dpop: "true",
+    },
+  });
 
 describe("TokenRefresher", () => {
   const defaultMocks = {
     storageUtility: StorageUtilityMock,
-    tokenRequester: TokenRequesterMock,
+    issuerConfigFetcher: mockIssuerConfigFetcher(mockDefaultIssuerConfig()),
+    clientRegistrar: mockDefaultClientRegistrar(),
   };
 
   function getTokenRefresher(
     mocks: Partial<typeof defaultMocks> = defaultMocks
   ): TokenRefresher {
     return new TokenRefresher(
       mocks.storageUtility ?? defaultMocks.storageUtility,
-      mocks.tokenRequester ?? defaultMocks.tokenRequester
+      mocks.issuerConfigFetcher ?? defaultMocks.issuerConfigFetcher,
+      mocks.clientRegistrar ?? defaultMocks.clientRegistrar
     );
   }
 
-  it("Refreshes the token properly", async () => {
-    const key = "global";
-    const refreshTokenValue = "refresh token value";
+  it("throws if no oidc issuer can be retrieved from storage", async () => {
+    const mockedStorage = mockStorageUtility({
+      "solidClientAuthenticationUser:mySession": {
+        codeVerifier: "some code verifier",
+        redirectUri: "https://my.app/redirect",
+        dpop: "true",
+      },
+    });
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(refresher.refresh("mySession")).rejects.toThrow(
+      "Failed to retrieve OIDC context from storage associated with session [mySession]"
+    );
+  });
+
+  it("throws if the token type cannot be retrieved from storage", async () => {
+    const mockedStorage = mockStorageUtility({
+      "solidClientAuthenticationUser:mySession": {
+        issuer: "https://my.idp",
+      },
+    });
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(refresher.refresh("mySession")).rejects.toThrow(
+      "Failed to retrieve OIDC context from storage associated with session [mySession]"
+    );
+  });
+
+  it("throws if a refresh token isn't provided", async () => {
+    setupDefaultOidcClientMock();
+    const mockedStorage = mockRefresherDefaultStorageUtility();
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(refresher.refresh("mySession")).rejects.toThrow(
+      "Missing a refresh token to refresh the access token associated to the session [mySession]"
+    );
+  });
+
+  it("throws if a dpop token is expected, but no dpop key is provided", async () => {
```

Comment:
```suggestion
  it("throws if a DPoP token is expected, but no DPoP key is provided", async () => {
```

Reply:
Resolved in 8818797c031a7e51ec3c56e1bf2e195d54ff6dee